Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp

```diff
@@ -178,6 +179,24 @@ namespace AZ
                     }
                 }
 
+                const AZStd::string resolutionModeCvarName("r_resolutionMode");
+                if (pCmdLine->HasSwitch(resolutionModeCvarName))
+                {
+                    auto numValues = pCmdLine->GetNumSwitchValues(resolutionModeCvarName);
+                    if (numValues > 0)
+                    {
+                        auto valueStr = pCmdLine->GetSwitchValue(resolutionModeCvarName);
+                        if (AZ::StringFunc::LooksLikeInt(valueStr.c_str()))
+                        {
+                            auto resolutionMode = AZ::StringFunc::ToInt(valueStr.c_str());
+                            if (resolutionMode >= 0)
+                            {
+                                r_resolutionMode = resolutionMode;
+                            }
+                        }
+                    }
+                }
```

Comment:
The AZ CVAR system automatically handles command line parameters.

Since this is an AZ CVAR handling the command line parameters is not needed.

I would also recommend using the callback parameter to [AZ_CVAR](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Console/IConsole.h#L211-L222) to handle updating the render resolution support if the CVar is changed while the application is running.

Reply:
> The AZ CVAR system automatically handles command line parameters.
> 
> Since this is an AZ CVAR handling the command line parameters is not needed.
> 
> I would also recommend using the callback parameter to [AZ_CVAR](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Console/IConsole.h#L211-L222) to handle updating the render resolution support if the CVar is changed while the application is running.

I believe these values are used before CVAR system is active and updated from the command line, hence the need to actively query the command line for any non-default values.

Reply:
https://github.com/o3de/o3de/commit/c1d56b73c63f01da92ad0bf7a0da85783bf5292c

It's definitely a good idea to make the cvar handle updates at runtime, but i'd suggest possibly saving that for a different PR, because the other related cvars (r_width, r_height, r_fullscreen) would all need to get updated for this.